frontend-ssr/pages/api/user/login.js

```diff
@@ -0,0 +1,38 @@
+//Imports
+const User = require("../../../models/User.schema");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+const { loginValidation } = require("../../../utils/validation/userValidation")
+
+//Vars
+const error_message = { status: "fail", message: 'Email or Password is incorrect' }
+
+export default async (req, res) => {
+    let { method, body } = { ...req };
+    switch (method) {
+        case "POST":
+            //destruct the req
+            let { email, password } = { ...body };
+            let { error } = loginValidation({ email, password })
+            //Check valid form 
+            if (error) return res.status(400).json(error_message);
+            //Check user base on username or email
+            let user = await User.findOne({ $or: [{ username: email }, { email: email }] });
+            if (!user) return res.status(400).json(error_message);
+            //User exists => Check password  
+            const validPass = await bcrypt.compare(password, user.password);
+            if (!validPass) return res.status(400).json(error_message);
+            //After checking password update the last login
+            let new_entry = await User.findOneAndUpdate({ _id: user._id }, { $set: { lastLogin: Date.now() } }, { new: true })
+            console.log(new_entry);
+            // Create jwt
+            const token = jwt.sign({ _id: user._id }, process.env.TOKEN_SECRET, { expiresIn: '7d' });
+            res.json({ status: "success", auth_token: token });
```

Comment:
Shouldn't we return the user data after login as well?

Reply:
I think this is more like a preference, I use to return just the `user._id` and check it if necessary, but we can return the full user in the JWT